### PR TITLE
Comprehensive reference and guide

### PR DESCRIPTION
**Pull request by masiedu4**
## Brief
- Update README file

## Activities
- This PR sets to provide documentation for the deroll project and its router and wallet functionalities

### CODE REVIEW

**Review by masiedu4 — COMMENTED** [review]
@tuler review requested

File: README.md [hunk 1]
```diff
@@ -114,20 +127,690 @@ validator-1  | [INFO  actix_web::middleware::logger] 127.0.0.1 "POST /finish HTT
 validator-1  | Hello world!
 ```
 
-Now you're ready to start building your Cartesi application with cartesi and deroll!
+## API and Usage Guide
+
+### 1. Create a new application
+
+#### `createApp(config: object): App`
```
Comment (tuler) [inline README.md:134]: for typescript developers, it's not only an object, it's a `HttpAppOptions`

File: README.md [hunk 2]
```diff
@@ -114,20 +127,690 @@ validator-1  | [INFO  actix_web::middleware::logger] 127.0.0.1 "POST /finish HTT
...
+    url: process.env.ROLLUP_HTTP_SERVER || "http://127.0.0.1:5004",
+  });
+
+  app.start();
+  ```
+  ---
+
+### 2. Create notices, vouchers and reports
+
+#### `createNotice(request: { payload: 0x${string}; }): Promise<number>`
```
Comment (tuler) [inline README.md:160]: This is actually a method of the `App` interface.
For a reference manual I'd rather have something generated.

File: README.md [hunk 3]
```diff
@@ -114,20 +127,690 @@ validator-1  | [INFO  actix_web::middleware::logger] 127.0.0.1 "POST /finish HTT
...
+  - **Throws:**
+    - `Error`: Throws an error if the response does not contain data or if the response status is not `OK`.
+
+  **Example:**
+
+  ```ts
+  const sample = "0x48656c6c6f2043617274657369"; // "Hello Cartesi" in hex
+
+  app
+    .createNotice({ payload: sample })
```
Comment (tuler) [inline README.md:180]: I prefer `async` syntax instead of `then/catch`

**Review by tuler — COMMENTED** [review]
Thanks for the PR.

I think most of the changes should be part of the documentation generated from typescript code.

I didn't have time to setup that generation yet.

File: README.md (hunk 2 above)
Comment (masiedu4) [reply README.md]: hi @tuler I pushed some changes

For the auto-generation, are we looking at something like TypeDoc? need some elaborate feedback there. thanks!

**Comment by tuler** [comment]
I configured typedoc myself, using the monorepo guidelines described at https://typedoc.org/api/#md:monorepos--workspaces. I even tried the upcoming [beta version](https://github.com/TypeStrong/typedoc/issues/2567), which improves support for monorepos, among other things.

Documentation is generated ok, but I don't like the result very much.
I'll try next week to follow another strategy, which is using VitePress for a docs website, and generating markdown from JSDoc into a VitePress website.

If that doesn't work well either, I'll reevaluate the options.

**Comment by tuler** [comment]
Documentation already published at https://deroll.dev

**Pull request closed without merging**